lib/renderer/bridge/session.go

```diff
@@ -83,8 +98,38 @@ func (s *session) Callback(data string) error {
 }
 
 func (s *session) evalJS(js string, mtype messageType) error {
-
 	// Prepend message type to message
 	return s.sendMessage(mtype.toString() + js)
+}
+
+// Shutdown
+func (s *session) Shutdown() {
+	s.done = true
+	s.shutdown <- true
+	s.log.Debugf("session %v exit", s.Identifier())
+}
 
+// writePump pulls messages from the writeChan and sends them to the client
+// since it uses a channel to read the messages the socket is protected without locks
+func (s *session) writePump() {
+	s.log.Debugf("Session %v - writePump start", s.Identifier())
+	for {
+		select {
+		case msg, ok := <-s.writeChan:
+			s.conn.SetWriteDeadline(time.Now().Add(1 * time.Second))
+			if !ok {
+				s.log.Debug("writeChan was closed!")
+				s.conn.WriteMessage(websocket.CloseMessage, []byte{})
+				return
+			}
+
+			if err := s.conn.WriteMessage(websocket.TextMessage, []byte(msg)); err != nil {
```

Comment:
we can now use `msg` directly, avoiding memory allocations on each connection.

Reply:
Or alternatively, just use `*(*[]byte)(unsafe.Pointer(&msg))` in the `WriteMessage` call :-)